clients/typescript/src/cli/migrations/migrate.ts

```diff
@@ -318,7 +329,37 @@ async function pascalCaseTableNames(prismaSchema: string): Promise<void> {
       //       such that we don't break indices
       //       when we will start inserting in the array later
       mapAnnotations = [[idx + 1, `  @@map("${tableName}")`], ...mapAnnotations]
+      replacements.set(tableName, modelName)
+    }
+  })
+
+  // Go over the schema again but now
+  // replace references to the old table names
+  // by the new model name when we are inside
+  // the definition of a model
+  let insideModel = false
+  lines = lines.map((ln) => {
+    if (ln.trim().startsWith('model ')) {
+      insideModel = true
+      return ln
+    }
+
+    if (insideModel && ln.trim().startsWith('}')) {
+      insideModel = false
+      return ln
     }
+
+    if (insideModel) {
+      // the regex below matches two identifiers separated by a space
+      // (first identifier is the column name, second one is its type)
+      const reg = /(\s*\w+\s+)(\w+)/
```

Comment:
```suggestion
      const reg = /^(\s*\w+\s+)(\w+)/
```

This should probably be applied only at the start of any given string